### DISCUSSION TITLE
Improve UI when working with Actions and Status Checks

### DISCUSSION THREAD

**Discussion by krystofspiller** [community/community#12511] [Actions]
Recently, I have came across two places in the GitHub UI that were related to Actions and weren't optimally done:

1. Scheduled reminders, i.e. `https://github.com/settings/reminders/<organization>` where the option "Your PR has failed CI checks" shows a free-text field to specify names of the check names without any autocompletion. It is also not obvious whether there is any way how to specify "all" checks. At least it is not possible to continue with an empty field which is not the case in the next scenario.
2. Branch protection rules, i.e. `https://github.com/<organization>/<repo>/settings/branch_protection_rules/new` where only checking the option "Require status checks to pass before merging" seemingly has no effect as for this to have an effect, the specific status checks need to be selected. Here, an autocompletion is available and it is possible to search for a check by name. However, it is also possible to leave it completely empty - in this case, I believe just checking the checkbox has no effect which is non-sensical and not at all clear. The same problem as in the previous case that it is not possible to specify "all" checks remains. It would also be great if a _workflow_ instead of individual _actions_ could be targeted. For example, we run our tests in parallel using the matrix strategy on 4 nodes. This spawns 4 different actions. Instead of needing to select 4 different actions, I would like to select the tests workflow. This also prevents issues in the future where we can for example scale this up to 8 nodes instead and the 4 new actions would not be required.

There is therefore chance for improvement on unification of the UI and further improvements. The final solution should in my opinion:
* Always provide autocompletion / search functionality.
* Provide a way to specify "all" checks. Maybe provide some wildcard (`*`) functionality.
* Always disallow continuing when no checks are specified.
* Provide a way to target workflows instead of specific actions (I believe this is what most people are after in the first place anyway).

**Comment by ethomson** [comment]
Hi @Laegas - sorry that you feel that this isn't optimal. We really appreciate you taking the time to give us detailed feedback about the gaps that you've identified and suggest some improvements! I'll talk this over with the team and make sure that we have improvements on the backlog.

Thanks again.

**Comment by CraigSiemens** [comment]
The text field is really what complicates it
- it's difficult to find out what format it's looking for
- you have to remember to add a new check here after it's added to a repo.

I think the options couple be simplified. Start out with just

- [ ] Your PR has failed CI checks

Then if that's checked, then show another option to limit it to required checks only.

- [x] Your PR has failed CI checks
  - [ ] Only required checks

**Comment by maximevaillancourt** [comment]
May I suggest writing some examples of expected values next to the input field? Even better would be a full documentation page somewhere on docs.github.com.

But maybe just to start, dropping a few informal examples in this discussion thread would be extremely helpful 🙏

**Reply by Gruebes** [reply]
Hope this helps: https://github.com/orgs/community/discussions/23986#discussioncomment-6826037

**Comment by tylerlaprade** [comment]
Any response? I see that the Microsoft rep who previously answered is no longer at the company. What's the best way to get visibility on this issue from current employees?

**Comment by magus424** [comment]
This continues to be a problem.
